internal/backend/backend.go

```diff
@@ -214,14 +232,28 @@ func newMetrics(registry prometheus.Registerer) *metrics {
 	return m
 }
 
-func (m *metrics) recordFailure(ctx context.Context, code int, uid types.UID) {
+func (m *metrics) recordFailure(ctx context.Context, code int, obj client.Object) {
+	log := log.FromContext(ctx)
+
+	if obj.GetUID() == "" {
```

Comment:
Instead of relying on the "internal" knowledge that the UID is empty on deletes, we could also require the caller to pass in the `deletedAt` timestamp variable, which is passed in on the `Upsert` call and check on that.

I'm not sure what's better to be honest, requiring the `deletedAt` timestamp feels a bit weird from an API-perspective as well. 

Opinions?

Reply:
I know what you mean about relying on the UID being empty... API-wise, using a deletedAt timestamp pointer (a golang `*time.Time` and using nil to express "not deleted" feels quite idiomatic, at least in Go...

I don't have strong opinions, and I'll approve this anyway. Feel free to change it or not 😁

Reply:
Alright I'll change it 😄